components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java

```diff
@@ -965,7 +957,7 @@ public static OMElement buildOMElement(InputStream inputStream) throws Exception
         int month = cal.get(Calendar.MONTH) + 1;
 
         if (!period.equals("" + year + "-" + month)) {
```

Comment:
why do we need an empty string to be concatinated with the value of "year"? Also, as a best practice, compare known constants against variables in conditional statement, which would help you avoid redundant null checks to see if the variable carries null.